hw/ip/spi_host/doc/_index.md

```diff
@@ -0,0 +1,633 @@
+This IP only supports single transfer rate (STR), *not* DTR.
+A preliminary investigation of DTR transfer mode suggests that proper support for setup and hold times in this mode may require a level of sub-cycle timing control which is not currently planned for this IP.
+
+# Theory of Operations
+
+## SPI_HOST IP Command Interface
+
+Issuing a command generally consists of the following steps:
+1. Configure the IP to be compatible with the target peripheral, using the {{< regref "CONFIG" >}} register,
+2. Load the instructions and data to be transmitted into the TXFIFO via the {{< regref "TXDATA" >}} register,
+3. Issue a 32-bit command to the {{< regref "COMMAND" >}} register with the following information,
+    - The speed to use for this particular command (Standard, Dual, or Quad)
+    - The three bit chip-select ID of the peripheral to target
+    - The number of bytes to transmit at *reduced* data rate, using `sd[0]` only (Dual or Quad rate *only*)
+        - This field is specifically for Dual or Quad commands where the first instruction must be sent serially on `sd[0]`
+    - The number of bytes to transmit at full data rate
+        - In Standard mode this controls the number of bytes transmitted and the number of bytes received, since the bus is full-duplex in Standard mode.
+    - The number of dummy `sck` cycles to wait before reading data (Dual or Quad rate *only*)
+    - The number of bytes to receive at the full Dual or Quad rate (Dual or Quad rate *only*)
+4. and read the peripheral response from the {{< regref "RXDATA" >}} register.
+
+### Configuration options
+
+In general, the {{< regref "CONFIG" >}} register holds clock configuration and timing settings which are specific to a particular class of target device.
+This register should be updated whenever a target peripheral requires different timing parameters than the device selected in the previous command.
+If all of the devices attached to the SPI_HOST have the same clock polarity, phase and timing requirements, the {{< regref "CONFIG" >}} register can in principle be initialized once and left unchanged for all subsequent commands.
+
+One of the most common differences between target devices is the requirement for a specific SPI clock phase or polarity, CPOL and CPHA, which were described in the previous section [SPI Protocol Basics](#spi-protocol-basics).
+These clock parameters can be set via the {{< regref "CONFIG.CPOL">}} or {{< regref "CONFIG.CPHA" >}} register fields.
+Likewise, as also described in the previous section, if device setup times require a full clock cycle before sampling the output, Full-Cycle Mode can be enabled by asserting the {{< regref "CONFIG.FULLCYC" >}} bit.
+
+#### Clock rate selection
+
+The SPI clock rate for each peripheral is set by two factors:
+- The SPI_HOST core clock (which may be generated independently from the TL-UL iterface clock)
+- An additional 16-bit clock divider
```

Comment:
are you thinking about actually dividing down to a new clock? Or are you thinking about just counting using source clock and finding the edges?

Reply:
No, just using an enable signal that is high once every N cycles.  Not a real divided down clock.  (Same thing as is done in pattgen).